suite-native/module-add-accounts/src/hooks/useAddCoinAccount.ts

```diff
@@ -0,0 +1,198 @@
+            },
+        });
+
+    const showAnotherEmptyAccountAlert = () =>
+        showAlert({
+            title: translate('moduleAddCoinAccount.alerts.anotherEmptyAccount.title'),
+            description: translate('moduleAddCoinAccount.alerts.anotherEmptyAccount.description'),
+            icon: 'warningCircle',
+            pictogramVariant: 'red',
+            primaryButtonTitle: translate(
+                'moduleAddCoinAccount.alerts.anotherEmptyAccount.actionPrimary',
+            ),
+            onPressPrimaryButton: () => {
+                hideAlert();
+            },
+            secondaryButtonTitle: translate(
+                'moduleAddCoinAccount.alerts.anotherEmptyAccount.actionSecondary',
+            ),
+            onPressSecondaryButton: () => {
+                openLink(
+                    translate('moduleAddCoinAccount.alerts.anotherEmptyAccount.actionSecondaryUrl'),
+                );
+                hideAlert();
+            },
+        });
+
+    const addCoinAddress = ({ network, type }: { network: Network; type?: AccountType }) => {
+        if (!device?.state) {
+            return false;
+        }
+
+        const accountType = type ?? getDefaultAccountType({ network });
+
+        const currentAccountTypeAccounts = accounts.filter(
+            account => account.symbol === network.symbol && account.accountType === accountType,
+        );
+
+        // Do not allow adding more than 10 accounts of the same type
+        if (currentAccountTypeAccounts.length > LIMIT) {
+            showTooManyAccountsAlert();
+            return false;
+        }
+
+        // Do not allow adding another empty account if there is already one
+        const emptyAccounts = currentAccountTypeAccounts.filter(account => account.empty);
+
+        if (emptyAccounts.length > 0) {
+            showAnotherEmptyAccountAlert();
+            return false;
+        }
+
+        const addNetworkAccount = async (deviceState: string) => {
+            setIsAddingCoinAccount(true);
+            const result = await dispatch(
+                addNetworkAccountThunk({
+                    network,
+                    accountType,
+                    deviceState,
+                }),
+            ).unwrap();
+
+            setIsAddingCoinAccount(false);
+
+            if (result.account) {
+                navigation.navigate(RootStackRoutes.AccountDetail, {
+                    accountKey: result.account.key,
+                    tokenContract: undefined,
+                });
+            }
+        };
+
+        addNetworkAccount(device.state);
+    };
+
+    const onSelectedNetworkItem = (networkSymbol: NetworkSymbol) => {
+        if (isAddingCoinAccount) {
+            return;
+        }
+        const network = getNetwork({ networkSymbol });
+        if (network) {
+            addCoinAddress({ network });
+        } else {
+            console.warn('Network not found', networkSymbol);
+        }
+    };
+
+    return {
```

Comment:
Why do you return all these values? `usAddCoinAccountScreen` uses only `supportedNetworkSymbols, onSelectedNetworkItem` should not it be enough?

Reply:
Good catch, some of them wont be used now. I'll remove them. Some will be used in following PR

Reply:
c333f3488a

Reply:
There is still many unused values in the object. Should not be there only `supportedNetworkSymbols` and `onSelectedNetworkItem`?

Reply:
Oh, you mentioned it in another thread that it will be used later... resolving.